spec/models/spree/gateway/shopify_gateway_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'spec_helper'
+
+module Spree
+  RSpec.describe Gateway::ShopifyGateway do
+    let(:transaction_id) { '0xDEADBEEF' }
+    let(:pos_order_id) { '0xBAADF00D' }
+    let(:refund) { double('refund', pos_order_id: pos_order_id) }
+    let(:refund_reason) { double('refund_reason', name: 'Product not working') }
+    let(:gateway_options) { { originator: refund } }
+
+    let(:provider_class) { ActiveMerchant::Billing::ShopifyGateway }
+    let(:provider_instance) { double('provider', refund: true, void: true) }
+
+    subject(:gateway) do
+      described_class.create!(name: "Shopify")
+    end
+
+    before do
+      gateway.preferences = { api_key: ENV['SHOPIFY_API_KEY'],
+                              password: ENV['SHOPIFY_PASSWORD'],
+                              shop_name: ENV['SHOPIFY_SHOP_NAME'] }
+
+      allow(provider_class).to receive(:new).and_return(provider_instance)
+    end
+
+    describe "#payment_profiles_supported?" do
+      it { is_expected.to_not be_payment_profiles_supported }
+    end
+
+    context '.void' do
+      let(:payments) { double(:payments, find_by: payment) }
+      let(:payment) { double(:payment, pos_order_id: pos_order_id) }
+
+      before do
+        allow(gateway).to receive(:payments).and_return(payments)
+      end
+
+      it 'calls the provider void method once' do
+        expect(provider_instance).to receive(:void).with('0xDEADBEEF', order_id: '0xBAADF00D')
+
+        void!
+      end
+
+      context "without a matching payment" do
+        let(:payment) { nil }
+
+        it 'raises an exception' do
+          cause = -> { void! }
```

Comment:
We 'should'. This is specifically why I was looking for your mental model around the lambda use: so I could fall in line.